tsinfer/formats.py

```diff
@@ -2007,30 +2064,69 @@ def num_alleles(self, sites=None):
     def variants(self, sites=None):
         """
         Returns an iterator over the Variant objects. This is equivalent to
-        the TreeSequence.variants iterator.
+        the TreeSequence.variants iterator. The alleles are guaranteed to be
+        returned in an order such that the ancestral state (if know) is the
+        first item in the list (and hence a ``0`` in the genotypes array) corresponds
+        to a sample with the ancestral state. This may mean that the indexes in the
+        genotypes array are recoded compared to the array passed in to :meth:`add_site`.
+
+        :param array sites: A numpy array of ascending site ids for which to return
+            data. If None (default) return all sites.
+        :return: An interator over the variants in the sample data file.
+        :rtype: iter(Variant)
+
         """
         all_genotypes = chunk_iterator(self.sites_genotypes, indexes=sites)
-        for genotypes, site in zip(all_genotypes, self.sites(ids=sites)):
-            yield Variant(site=site, alleles=site.alleles, genotypes=genotypes)
+        ancestral_allele_index = self.sites_ancestral_allele_index[:]
```

Comment:
Don't need to do this, as the ``site`` returns ``ancestral_allele_index``.

Within the loop:
```python
alleles = site.alleles
if recode_ancestral:
    alleles = site.remap_alleles()
    genotypes = site.remap_genotypes(genotypes)
# NOTE: if recode_ancestral is True then variant.alleles and site.alleles may differ.
yield Variant(site=site, alleles=alleles, genotypes=genotypes)
```